src/main/java/io/stargate/sgv2/jsonapi/service/resolver/matcher/TableFilterResolver.java

```diff
@@ -104,7 +103,7 @@ private static List<DBFilterBase> findDynamic(CaptureExpression captureExpressio
                 captureExpression.path(),
                 NativeTypeTableFilter.Operator.from(
                     (ValueComparisonOperator) filterOperation.operator()),
-                (BigDecimal) rhsValue));
+                (Number) rhsValue));
```

Comment:
Just a note - the changes yuqi and I are doing will revert how the resolver functions here are called back to the original approach where it gets called once with everything that matched. Currently it is getting called for every match. 

Nothing to change, just a note that when we do that we should look at how to reduce the use of instanceof and casting

Reply:
Thank you: that sounds like a good improvement.